dask/array/routines.py

```diff
@@ -332,6 +333,52 @@ def vdot(a, b):
     return dot(a.conj().ravel(), b.ravel())
 
 
+def _shape_minus_axis(shape, axis):
+    newshape = list(shape)
+    newshape.pop(axis)
+    return tuple(newshape)
+
+
+def _chunk_sum(a, axis=None, dtype=None, keepdims=None):
+    # Caution: this is not your conventional array-sum: due
+    # to the special nature of the preceding blockwise con-
+    # traction,  each chunk is expected to have exactly the
+    # same shape,  with a size of 1 for the dimension given
+    # by `axis` (the reduction axis).  This makes mere ele-
+    # ment-wise addition of the arrays possible.   Besides,
+    # the output can be merely reshaped to lose the `axis`-
+    # dimension when keepdims = False
+    if type(a) is list:
+        xp = np
+
+        if is_cupy_type(a[0]):
+            import cupy
+
+            xp = cupy
```

Comment:
Why is this needed? wouldn't numpy handle the dispatch to cupy below?

Reply:
Honestly, I do not know.  But the exact same construct is being used in `_matmul()` (which you allowed to remain), which is why I copied it over here.

Reply:
@ParticularMiner that part actually came after my change via https://github.com/dask/dask/pull/7563. @quasiben would appreciate your input here? When is that `is_cupy_type` needed?

Reply:
Sorry I assumed it was yours @ravwojdyla

Reply:
`is_cupy_type` is normally needed when we need to support the non-NumPy API part of CuPy, such as `cupyx.scipy`, this is why `matmul` requires that check. I think the reduction here may not need the same, or at least I can't think of a need for that right now. I would then suggest just replacing all this by `np.add` and it should work. If gpuCI tests pass we should be good to what we know is supported today.

@quasiben please feel free to correct me.

Reply:
Sorry @pentschev , I didn't see your comment above until just now.  Nevertheless, I've already taken steps that agree with your comment.